src/QMCDrivers/QMCDriverNew.cpp

```diff
@@ -87,15 +87,17 @@ QMCDriverNew::QMCDriverNew(const ProjectData& project_data,
 
 QMCDriverNew::~QMCDriverNew() = default;
 
-void QMCDriverNew::checkNumCrowdsLTNumThreads(const int num_crowds)
+int QMCDriverNew::determintNumCrowds(const int requested_num_crowds, const int rng_count)
 {
-  int num_threads(Concurrency::maxCapacity<>());
-  if (num_crowds > num_threads)
+  int num_crowds = requested_num_crowds;
+  if (requested_num_crowds == 0)
+    num_crowds = rng_count;
+  else if (requested_num_crowds > rng_count)
   {
-    std::stringstream error_msg;
-    error_msg << "Bad Input: num_crowds (" << num_crowds << ") > num_threads (" << num_threads << ")\n";
-    throw UniformCommunicateError(error_msg.str());
+    app_warning() << "Capping the number of crowds to the count of driver-captured RNGs : " << rng_count << std::endl;
+    num_crowds = rng_count;
```

Comment:
Small point: Warning should indicate how this might have happened + possible user fix. How could it happen?

Reply:
>As I read the current code a scenario were OMP_NUM_THREADS < num_crowds. at the moment. Which to me is pretty undesirable, but of course I can't think of a sane reason that num_crowds > num threads possible. So personally I think we should halt but we should definitely direct the user that they must set OMP_NUM_THREADS >= num_crowds.

With this PR, `OMP_NUM_THREADS < num_crowds` can run in a unit test as long as enough RNGs are captured in the driver. This is not the case with the old code with directly check against OMP_NUM_THREADS.

>I don't think we should be setting the rank global number of OMP threads with an environment variable. I think there should be a formal global element in the input for this. It's not portable to future threading system support and it leaves a very important input element invisible in input.

Process affinity is always controlled outside the application. If we handle the number of threads via input, it is another parameter to control and I don't feel such a knob brings value to us. Generally, I prefer all the resource (core/GPU) affinity handling by a single place `mpirun`.

>The max threads call is also problem in that it doesn't nessarily carry the meaning of the max threads the current process is allowed. std::thread and other libraries appear to have a call like that but the underlying thread mechanism doesn't always provide a sensible response pthreads gives you a number that is either maximum number of threads the kernel can handle or just the max value of a type.

pthread std::thread lack the concept of thread pool and affinity control. That is why openmp runtime was introduced on top of pthread and got good adoption.

Reply:
Thanks for addressing my comment.